File: services/networkx_service/src/process/algorithm/shortest_path/bellman_ford.py
import networkx as nx
from algorithm.algorithm import Algorithm


class BellmanFord(Algorithm):
    def __init__(self, instance, publishers):
        super().__init__(instance, publishers)
        self.source = self.parameters['source']
        self.target = self.parameters['target']

    def algorithm(self):
        path = nx.bellman_ford_path(self.graph, int(self.source), int(self.target))

        return len(path) - 1

File: services/graph4j_service/src/process/src/main/java/process/algorithm/shortestpath/BellmanFord.java
package process.algorithm.shortestpath;

import org.graph4j.alg.sp.BellmanFordShortestPath;
import org.graph4j.util.Path;
import process.algorithm.Algorithm;
import process.instance.Instance;
import process.pubsub.Publisher;

import java.util.List;

public class BellmanFord extends Algorithm {
    Integer source;
    Integer target;

    public BellmanFord(Instance instance, List<Publisher> publishers) {
        super(instance, publishers);
        this.source = parameters.get("source").toInt();
        this.target = parameters.get("target").toInt();
    }

    public String algorithm() {
        BellmanFordShortestPath bellmanFordShortestPath = new BellmanFordShortestPath(this.graph, this.source);
        Path path = bellmanFordShortestPath.findPath(this.target);

        return String.valueOf(path.length());
    }
}
